internal/internal_activity.go

```diff
@@ -311,32 +317,35 @@ func deSerializeFnResultFromFnType(fnType reflect.Type, result []byte, to interf
 		if result == nil {
 			return nil
 		}
-		err := getHostEnvironment().decodeArg(result, to)
+		err := decodeArg(dataConverter, result, to)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func deSerializeFunctionResult(f interface{}, result []byte, to interface{}) error {
+func deSerializeFunctionResult(f interface{}, result []byte, to interface{}, dataConverter encoded.DataConverter) error {
 	fType := reflect.TypeOf(f)
+	if dataConverter == nil {
```

Comment:
This is actually on purpose, for test code that use this, I want people just pass nil instead of figuration out how to deal with dc. So prefer not change

Reply:
Understand, that's what I already did. But to be in defensive way, I feel it is still better to have this check, especially in this function.